airbyte-webapp/src/packages/cloud/views/credits/CreditsPage/components/UsagePerConnectionTable.tsx

```diff
@@ -19,24 +21,44 @@ const Content = styled.div`
 type UsagePerConnectionTableProps = {
   creditConsumption: CreditConsumptionByConnector[];
 };
+
+type FullTableProps = CreditConsumptionByConnector & {
+  creditsConsumedPercent: number;
+  sourceIcon: string;
+  destinationIcon: string;
+};
+
 const UsagePerConnectionTable: React.FC<UsagePerConnectionTableProps> = ({
   creditConsumption,
 }) => {
   const { query, push } = useRouter();
+  const { sourceDefinitions } = useSourceDefinitionList();
+  const { destinationDefinitions } = useDestinationDefinitionList();
 
   const creditConsumptionWithPercent = React.useMemo(() => {
     const sumCreditsConsumed = creditConsumption.reduce(
       (a, b) => a + b.creditsConsumed,
       0
     );
-    return creditConsumption.map((item) => ({
-      ...item,
-      creditsConsumedPercent:
-        item.creditsConsumed !== 0
-          ? (item.creditsConsumed / sumCreditsConsumed) * 100
-          : 0,
-    }));
-  }, [creditConsumption]);
+    return creditConsumption.map((item) => {
+      const currentSourceDefinition = sourceDefinitions.find(
+        (def) => def.sourceDefinitionId === item.sourceDefinitionId
+      );
+      const currentDestinationDefinition = destinationDefinitions.find(
+        (def) => def.destinationDefinitionId === item.destinationDefinitionId
+      );
+
+      return {
+        ...item,
+        sourceIcon: currentSourceDefinition?.icon,
+        destinationIcon: currentDestinationDefinition?.icon,
+        creditsConsumedPercent:
+          item.creditsConsumed !== 0
+            ? (item.creditsConsumed / sumCreditsConsumed) * 100
```

Comment:
What if `sumCreditsConsumed` is 0? I think we could change it to `sumCreditsConsumed ? item.creditsConsumed / sumCreditsConsumed * 100 : 0`